### PR TITLE
NNMF wrapper

### PR DESCRIPTION
**Pull request by topepo**
Updated from your develop branch

### CODE REVIEW

File: R/nnmf.R [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#' proj_dat <- factorization@apply(dat)
+#'
+#' plot(proj_dat@data[, 1], proj_dat@data[, 2])
+#'
+#' @include dimRedResult-class.R
+#' @include dimRedMethod-class.R
+#' @family dimensionality reduction methods
+#' @export NNMF
+#' @exportClass NNMF
+#' @importFrom NNLM nnmf nnlm
```
Comment (gdkrmr) [inline R/nnmf.R]: I put all the packages `dimRed` depends on in the `Suggests` section of the `DESCRIPTION` file and I don't import anything directly if I can avoid it. There is `chckpkg` to check for dependencies. 
`dimRed` depends on many packages and this way I avoid problems with CRAN if one of these packages should ever get removed from CRAN.

Adding NNLM to DESCRIPTION/Suggests and removing this line should also get you through the first set of `R CMD check` tests on travis.

**Comment by topepo** [comment]
I've updated it to move from Imports to Suggests.

**Comment by gdkrmr** [comment]
ok, great, does it work on your computer? Travis currently complains:
```
1. Error: Bad args (@test_NNMF.R#188) 
invalid class "dimRedData" object: data must be numeric
1: expect_failure(embed(iris, "NNMF")) at testthat/test_NNMF.R:188
2: capture_expectation(expr)
3: tryCatch({
       code
       NULL
   }, expectation = function(e) e)
4: tryCatchList(expr, classes, parentenv, handlers)
5: tryCatchOne(expr, names, parentenv, handlers[[1L]])
6: doTryCatch(return(expr), name, parentenv, handler)
7: embed(iris, "NNMF")
8: embed(iris, "NNMF")
9: .local(.data, ...)
10: embed(as(.data, "dimRedData"), .method, .mute, .keep.org.data, ...)
11: as(.data, "dimRedData")
12: asMethod(object)
13: new("dimRedData", data = as.matrix(from))
14: initialize(value, ...)
15: initialize(value, ...)
16: .local(.Object, ...)
17: callNextMethod()
18: .nextMethod(.Object = .Object, data = data)
19: validObject(.Object)
20: stop(msg, ": ", errors, domain = NA)
```
Does it run on your computer?

**Comment by topepo** [comment]
It does now :-)

`R CMD check` was clean except that that I couldn't get that damn `pcaL1` package to install. I guess it didn't run that particular test.

**Comment by gdkrmr** [comment]
merged, added some goodies: Inverse function and moved W to the actual results. Thanks again!

**Pull request merged**